### ISSUE TITLE
Help Required in User's Device Information

### ISSUE THREAD

**Issue by jmandivarapu1** [aws-amplify/amplify-js#4635]
** Which Category is your question related to? **
E.g. Auth
** What AWS Services are you utilizing? **
E.g. Cognito
 I am trying to check if the user's last device is equal to the current device he logged in.

So, I thought of using my Auth to get the current device information and got it using **Auth.currentSession()**. This gave me my device key and I assumed this key is unique per device(I am wrong I think). So when I logged out and login again. It gave me a new device_key. Even though on the same system and browser and user. 

So Is there any way to tell that user logged in from the same device?
```
accessToken: CognitoAccessToken
jwtToken: "eyJraWQiOiJ1dw"
payload:
sub: "ce110900-31ae-452a-ad13-c4c235029666"
device_key: "us-west-2_cbe09344-1ca9-4363-b566

```

**Comment by Amplifiyer** [comment]
@jmandivarapu1 just to confirm, you are checking the remembered device according to the APIs mentioned here https://aws.amazon.com/blogs/mobile/tracking-and-remembering-devices-using-amazon-cognito-your-user-pools/?

**Comment by jmandivarapu1** [comment]
> @jmandivarapu1 just to confirm, you are checking the remembered device according to the APIs mentioned here https://aws.amazon.com/blogs/mobile/tracking-and-remembering-devices-using-amazon-cognito-your-user-pools/?

Yes. Mine is "always" to remember users devices

**Comment by jmandivarapu1** [comment]
Any update on this if someone knows annything?

**Comment by jmandivarapu1** [comment]
Is this github of Amplify Active or not?

**Comment by jmandivarapu1** [comment]
Does anybody know answer to this?

**Comment by Amplifiyer** [comment]
Hi @jmandivarapu1 sorry for missing this issue for this long. I tried to reproduce the issue and learn more about remembering the device. On the first login of the user on a device, you'll get a `DeviceKey` and `DeviceGroupKey` which is supposed to be persisted in the client (at user scope). Right after the first call, amplify will make another call `AWSCognitoIdentityProviderService.ConfirmDevice` to mark this device as remembered for the user. Signing out is supposed to not clear these two keys from the local storage (verified that they are not using both `withAuthenticator` HOC as well as `Auth.signIn` and `Auth.signOut` APIs)

Further sign in calls use the saved `DeviceKey` and `DeviceGroupKey` as ChallengeResponses to `AWSCognitoIdentityProviderService.RespondToAuthChallenge` for `DEVICE_SRP_AUTH`

In summary, if you are not clearing our your local storage between multiple sign-in on the same device (app reinstalls, cognito mode in browsers etc), the feature should work as intended.
